static/usr/lib/core/rtc-sys-time-init

```diff
@@ -0,0 +1,112 @@
+#!/bin/sh
+
+# Adjust system time for the following reasons:
+#     1) Sync system time to RTC time
+#          - On older kernel versions, RTC modules with HTCOSYS enabled cannot sync system time with
+#            RTC time (attribute hctosys==0)
+#     2) Correct driver/module system time sync to invalid time (earlier than most recent time stamp)
+#          - Built-in/loadable RTC driver with HCTOSYS enabled may set invalid time due to .e.g
+#            uninitialized RTC or bad battery
+#
+# If the RTC time is not valid, or cannot be set, forward the system time to most recent timestamp
+
+set -eu
+
+journal() {
+    pri=$1
+    shift
+    lvl=
+    case "$pri" in
+        "error")
+            lvl="<3>"
+            ;;
+        "warning")
+            lvl="<4>"
+            ;;
+        "debug")
+            lvl="<7>"
+            ;;
+        *)
+            lvl="<6>"
+            ;;
+    esac
+
+    # Apply priority level to all lines
+    printf "%s\n" "$@" | while IFS= read -r line; do
+        printf "%s\n" "$lvl$line"
+    done
+}
+
+# Expects single argument RTC device node /dev/rtc* (instance name dev-rtc* with escaping undone)
+dev_node="/$1"
+
+# Get systemd timestamp file created during core build
+clock_epoch_stamp=0
+if [ -e "/usr/lib/clock-epoch" ]; then
+    clock_epoch_stamp="$(stat -L "/usr/lib/clock-epoch" -c '%Y')"
+else
+    journal warning "Warning: cannot stat /usr/lib/clock-epoch"
+fi
+clock_epoch_date=$(date -d @"$clock_epoch_stamp")
+journal debug "core/systemd build timetamp: $clock_epoch_date"
+
+# Get systemd-timesyncd timestamp file initialized to systemd build time and updated periodically
+# (default 60s) and at shutdown
+timesync_clock_stamp=0
+if [ -e "/var/lib/systemd/timesync/clock" ]; then
+    timesync_clock_stamp="$(stat -L "/var/lib/systemd/timesync/clock" -c '%Y')"
+else
+    journal warning "Warning: cannot stat /var/lib/systemd/timesync/clock"
```

Comment:
After we will have removed the broken hack in `901-cleanup-timesyncd.chroot`, this will still happen on the first boot. In order not to become a red herring, we should say that this may be expected on first boot in the warning message.

Reply:
Thanks, that is a good point, fixed.